### ISSUE TITLE
It takes a long time until the Login-Page appears

### ISSUE THREAD

**Issue by SimonSays4223** [TheHive-Project/TheHive#528]
# It takes a long time until the Login-Page appears

### Request Type
Bug

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| OS version (server)       |  CentOS 7.4.1708
| OS version (client)       | CentOS 6.9 (Final)
| TheHive version / git hash   | 3.0.6-1
| Package Type              | RPM
| Browser type & version    | Seamonkey (2.49.3) and Firefox (52.7.0)
| Java  | Java (OpenJDK) 1.8.0.161

### Problem Description
On first connect, the hive Login-Page appears only after a few minutes (the loading animation appears neither). We checked the SSL-Handshake with openssl connect s_client.. everything is okay, fast response. We have the same installation with Centos 6.9 and hive 3.0.1-1 and HTTPs. In this installation everything is working as expected. If we disable HTTPs in the CentOS 7 installation everything is working as expected as well. We think there must be an issue somewhere in between.

We also tried to...
- convert the jks to an pkcs12 - nothing changed
- create a new snakeoil jks - nothing changed
- inspect pcaps - nothing special
- tried to set different values for timeouts (Listing [2])
- ask on gitter

How can we debug this problem? We set everything to DEBUG in logback.xml but we would like to see more like in apache or ngninx with higher verbose levels. Are there some more -vvv or -d switches we can set?

Explanation Listing 1:
On "New Connection" until the last line in the Listing we have to wait until the Login-Page appears. We also tried to set the timer to 3s as described in Listing [2] in application.conf - nothing changed. Afterwards we set the timer explicitly with the -D option according to a Ticket in the play framework (#8286) and it worked.

### Steps to Reproduce
1. Install the environment as described above
2. Create a KeyStore file as described in the thehive manual
3. Enable HTTPs GUI as described in the thehive manual

For cross-checking we installed the same machine (ansible) in a different (virtual) server environment. Same Problem.

### Possible Solutions
If we start thehive manually and add -Dplay.http.server.idletimeout=3s we can force the login prompt to appear quicker - but this is a little bit dirty actually we try to avoid such hacks. This is not a solution.

### Complementary information
Listing [1]:
``Apr 05 12:25:54 OurMachineName thehive[1822]: [debug] a.i.TcpListener - Successfully bound to /0.0.0.0:9443
Apr 05 12:25:54 OurMachineName thehive[1822]: [info] p.c.s.AkkaHttpServer - Listening for HTTPS on /0.0.0.0:9443
Apr 05 12:25:56 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:21 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:41883 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:27:21 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:27:21 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:27:22 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:28:45 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:44430 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:28:45 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:44775 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:45166 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:29498 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:42148 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.a.RepointableActorRef - Aborting tcp connection to /OurMachinesIPAddr:16892 because of upstream failure: HTTP idle-timeout encountered, no bytes passed in the last 75000000000 nanoseconds. This is configurable by akka.http.[server|client].idle-timeout.
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.TLSActor - closing output
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.i.TcpListener - New connection accepted
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:28:46 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:28:47 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:28:47 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
Apr 05 12:28:47 OurMachineName thehive[1822]: [debug] a.s.i.i.InputStreamPublisher - No more bytes available to read (got `-1` from `read`)
``

Listing [2]:

- /etc/thehive/application.conf
- #akka.http.server.linger-timeout = 3s
- #akka.http.server.request-timeout = 3s
- #akka.http.server.bind-timeout = 1s
- #akka.http.server.idle-timeout = 3s
- #akka.http.client.idle-timeout = 3s
- #akka.http.host-connection-pool.idle-timeout = 3s
- #akka.http.host-connection-pool.client.idle-timeout = 3s

**Comment by SimonSays4223** [comment]
Yes it takes the same long time. After that i get a long string with version information.

**Comment by To-om** [comment]
Can you paste the output ? Or tell me if there is an error ?

**Comment by SimonSays4223** [comment]
I can't see any error. Here it comes:

``{"versions":{"TheHive":"3.0.6","Elastic4Play":"1.4.5","Play":"2.6.7","Elastic4s":"5.6.0","ElasticSearch":"5.6.2"},"connectors":{},"health":{"elasticsearch":"WARNING"},"config":{"protectDownloadsWith":"ThisIsAString","authType":["key","local"],"capabilities":["authByKey","changePassword","setPassword"]}}``

**Comment by To-om** [comment]
You confirm that you haven't any Cortex nor Misp configured in TheHive ?

Connection with ElasticSearch doesn't it work correctly ?

Please run the following command (replace ip address and index name if necessary) :
`curl http://127.0.0.1:9200/_cluster/health/the_hive_13?pretty`

**Comment by SimonSays4223** [comment]
First question:
I configured a cortex instance as described in the manual. I just removed the configuration lines restarted the thehive service - same result (takes a long time). 

Second question:
I think it is working correctly. Yes there is a Warning but it is just a Warning:
``
Cluster health status changed from [RED] to [YELLOW] (reason: [shards started[[the_hive12][1]] ..])
``
(This line above is not a copy i wrote it by hand so there may be something unimportant missing)

Your order:

[root@ThankYouTo-om]# curl '127.0.0.1:9200/_cluster/health/the_hive_13?pretty'
{
  "cluster_name" : "hive",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 5,
  "active_shards" : 5,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 5,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 50.0
}


[root@forYourHelp]# curl '127.0.0.1:9200/_cluster/health/the_hive_12?pretty'
{
  "cluster_name" : "hive",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 5,
  "active_shards" : 5,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 5,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 50.0

As you can see, i have two indices. I restored (according to the manual) an old "database" in this instance. (I didn't do that in the cross-check machine i set up to double-check the behavior).

**Comment by To-om** [comment]
If you've configured a Cortex, can you check that it is reachable from TheHive server.

Execute this command from TheHive: `curl http://cortex-ip/api/status`

**Comment by SimonSays4223** [comment]
I changed your command to: 

- curl http://127.0.0.1:!!9000!!/api/status. (without !!)

Due to firewalling it is/was not reachable (i am aware of this situation). Now i changed the rules and it is reachable and i get the following output: 

[root@myIncredibleTHMachine]# curl http://CortexIpAddress:9000/api/status
{"versions":{"Cortex":"1.1.4","Play":"2.5.9"},"config":{"authType":"none","capabilities":[]}}

Regarding to the access times nothing changed even with a working cortex connection.

Nevertheless i would like to omit cortex at this stage. In my opinion TH must work without this connection.

**Comment by To-om** [comment]
You confirm that even with a working connection to Cortex, request to `https://yourIncredibleTHMachine:9443/api/status` still long, as TheHive login page ?

I agree with you, login page must not take long time if Cortex, or other component, is unreachable.

**Comment by SimonSays4223** [comment]
Yes i confirm - even with a working cortex connection it takes a long time. How can i debug the connection in more detail?!

**Comment by SimonSays4223** [comment]
Remember... the installation with CentOS69 and the older TH version as descibed in the initial text works like a boss. I know what changed in CentOS6 to 7 or at least i am able to help myself very well but i have no clue what changed in TH and i cant help myself - my java skills are very rudimental except there is some c code.
